frontend/packages/console-plugin-sdk/src/typings/features.ts

```diff
@@ -4,42 +4,34 @@ import { Extension, AlwaysOnExtension } from './base';
 import { Action } from 'typesafe-actions';
 
 namespace ExtensionProperties {
-  interface FeatureFlag {
-    /** The name of the feature flag. */
+  export interface ModelFeatureFlag {
     flag: string;
-  }
-
-  export interface ModelFeatureFlag extends FeatureFlag {
     /** If a CRD for this model exists, the feature will be enabled. */
     model: K8sKind;
   }
 
-  export interface ActionFeatureFlag extends FeatureFlag {
-    /** Function used to detect the feature and set flag name/value via Redux action dispatch. */
-    detect: ActionFeatureFlagDetector;
+  export interface FeatureDetect {
+    /** Function used to detect the feature and set arbitrary flag name/value via Redux action dispatch. */
+    detect: ActionFeatureDetector;
   }
 }
 
 export interface ModelFeatureFlag extends AlwaysOnExtension<ExtensionProperties.ModelFeatureFlag> {
   type: 'FeatureFlag/Model';
 }
 
-export interface ActionFeatureFlag extends Extension<ExtensionProperties.ActionFeatureFlag> {
-  type: 'FeatureFlag/Action';
+export interface FeatureDetect extends Extension<ExtensionProperties.FeatureDetect> {
+  type: 'FeatureDetector/Action';
 }
 
-export type FeatureFlag = ModelFeatureFlag | ActionFeatureFlag;
+export type FeatureFlag = ModelFeatureFlag;
 
 export const isModelFeatureFlag = (e: Extension): e is ModelFeatureFlag => {
   return e.type === 'FeatureFlag/Model';
 };
 
-export const isActionFeatureFlag = (e: Extension): e is ActionFeatureFlag => {
-  return e.type === 'FeatureFlag/Action';
-};
-
-export const isFeatureFlag = (e: Extension): e is FeatureFlag => {
-  return isModelFeatureFlag(e) || isActionFeatureFlag(e);
+export const isFeatureDetector = (e: Extension): e is FeatureDetect => {
+  return e.type === 'FeatureDetector/Action';
 };
 
-export type ActionFeatureFlagDetector = (dispatch: Dispatch) => Promise<void | Action>;
+export type ActionFeatureDetector = (dispatch: Dispatch) => Promise<void | Action>;
```

Comment:
```suggestion
export type FeatureDetector = (dispatch: Dispatch) => Promise<void | Action>;
```

I'd skip the `Action` prefix here.